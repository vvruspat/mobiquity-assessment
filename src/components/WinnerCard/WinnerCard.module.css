.content {
    display: flex;
    flex-direction: row;
    justify-content: stretch;
    gap: var(--padding-medium);
}

.flag {
}

.info {
    display: flex;
    flex-direction: column;
    justify-content: stretch;
    align-items: flex-start;
    gap: var(--padding-small);
    flex: 1;
}

.title {
    color: var(--regular-color);
}

.team {
}

.stat {
}