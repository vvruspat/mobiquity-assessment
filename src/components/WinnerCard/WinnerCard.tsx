import { FC, PropsWithChildren } from "react";

import s from "./WinnerCard.module.css";
import { Winner } from "../../types/winner";
import { Card } from "../Card";
import { Text } from "../Text";
import { Flag } from "../Flag";
import { PointsBadge } from "../PointsBadge";
import { TeamBadge } from "../TeamBadge";

type WinnerCardProps = Winner;

/**
 * WinnerCard component
 */
export const WinnerCard: FC<PropsWithChildren<WinnerCardProps>> = ({
  year,
  name,
  points,
  team,
  nationality,
}) => {
  return (
    <Card isAccent={true} header={<Text size="m">Winner {year}</Text>}>
      <div className={s.content}>
        <div className={s.flag}>
          <Flag nationality={nationality} size="l" />
        </div>
        <div className={s.info}>
          <div className={s.title}>
            <Text size="xxl" weight="semibold">
              {name}
            </Text>
          </div>
          <div className={s.team}>
            <TeamBadge team={team} />
          </div>
        </div>
        <div className={s.stat}>
          <PointsBadge points={points} />
        </div>
      </div>
    </Card>
  );
};
