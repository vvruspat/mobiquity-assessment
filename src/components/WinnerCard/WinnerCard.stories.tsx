import { ComponentStory, ComponentMeta } from "@storybook/react";

import { WinnerCard } from "./WinnerCard";
import "../../index.css";

export default {
  title: "Components/WinnerCard",
  component: WinnerCard,
} as ComponentMeta<typeof WinnerCard>;

const Template: ComponentStory<typeof WinnerCard> = (args) => (
  <WinnerCard {...args}>Card content</WinnerCard>
);

export const CardTemplate = Template.bind({});

CardTemplate.args = {
  year: "2022",
  driver: "Charles Leclerc",
  points: 256,
  team: "RED BULL",
  nationality: "Spanish",
};
