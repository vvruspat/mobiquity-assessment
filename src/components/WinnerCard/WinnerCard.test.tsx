import { render, screen } from "@testing-library/react";
import { WinnerCard } from "./WinnerCard";

const props = {
  year: "2022",
  driver: "Charles Leclerc",
  points: 256,
  team: "RED BULL",
  nationality: "Spanish",
};

test("renders race card", () => {
  render(<WinnerCard {...props} />);

  const el = screen.getByTestId("card");
  expect(el).toHaveTextContent(`Winner ${props.year}`);
  expect(el).toHaveTextContent("Charles Leclerc");
  expect(el).toHaveTextContent("RED BULL");

  expect(el).toMatchSnapshot();

  const flag = screen.getByTestId("flag");
  expect(flag).toBeInTheDocument();
  expect(flag).toHaveClass("fi-es");
});
