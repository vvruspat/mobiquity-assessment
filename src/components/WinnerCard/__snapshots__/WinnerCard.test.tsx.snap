// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders race card 1`] = `
<section
  class="card card_accent"
  data-testid="card"
>
  <header
    class="header"
  >
    <span
      class="text size_m regular"
      data-testid="text"
    >
      Winner 
      2022
    </span>
  </header>
  <div
    class="content"
  >
    <div
      class="content"
    >
      <div
        class="flag"
      >
        <div
          class="flag flag_l fi fi-es"
          data-testid="flag"
        />
      </div>
      <div
        class="info"
      >
        <div
          class="title"
        >
          <span
            class="text size_xxl semibold"
            data-testid="text"
          >
            Charles Leclerc
          </span>
        </div>
        <div
          class="team"
        >
          <div
            class="badge badge_m badge_style_primary badge_align_center"
            data-testid="badge"
          >
            <div
              class="team"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                RED BULL
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stat"
      >
        <div
          class="badge badge_l badge_style_info badge_align_center"
          data-testid="badge"
        >
          <h6
            class="caption"
            data-testid="caption"
          >
            <span
              class="text size_s regular"
              data-testid="text"
            >
              Points
            </span>
          </h6>
          <span
            class="text size_xxl regular"
            data-testid="text"
          >
            256
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
